docs/internal/generics.md

```diff
@@ -0,0 +1,114 @@
+# Generics
+
+Generics are a way to use type parameters to define or restrict other types.
+For example, a generic class can use its type parameters to define the type of fields, and a generic method can use its parameters to restrict what can be passed as an argument.
+
+Example:
+```ts
+class MyGenericType[MyType, SomeInterface]
+{
+  // MyType is the type of this field
+  field : MyType;
```

Comment:
```suggestion
  var field : MyType;
```

I was debating whether to add an `& mut` in this suggestion.  If we don't add it, then the generic parameter must contain the capability types as well as the concrete type, if we do then the parameter can't specify something different.